### ISSUE TITLE
Improve adding media upload to a block

### ISSUE THREAD

**Issue by mkaz** [WordPress/gutenberg#22273]
**Is your feature request related to a problem? Please describe.**

When creating a new block that requires a media upload, there is a fair amount of complexity to get it right. You can see this in the core "Media & Text" block which ends up creating its own `MediaContainer`, to include the `MediaPlaceholder` and lots of additional functions to account for image vs video, resizing, blobUrl replacement and other aspects that go into an upload.

Additionally, even with the additional code in that block, an upload in-progress indicator is still not included. Digging through the Image block, shows the indicator is handled using a select data and mediaUpload function, though a fair amount of additional complexity that everyone who needs to create an upload will need to duplicate.

**Describe the solution you'd like**

Ideally, we create a single component for upload that is easy to use and requires a minimal of customization to get all the pieces to upload. This includes indicator, drag-drop, blobUrl replacement, etc...

**Comment by chrisvanpatten** [comment]
I agree completely — the APIs here are pretty tricky to use. There's a lot of room for cleaning up this experience.

**Comment by talldan** [comment]
I expect this is still a challenge today, though maybe slightly easier thanks to some better components that have more built in (`MediaReplaceFlow`).

I think some of what was mentioned in [Block Attribute Groups](https://github.com/WordPress/gutenberg/issues/73845#top) and [Pattern Editing and contentOnly interactivity](https://github.com/WordPress/gutenberg/issues/71517#top) might be an avenue that makes this easier, possibly in the future, if a media attributes are configured correctly, a user will be able to opt in to the default media controls (which perhaps might not include the placeholder, but I would imagine the appropriate toolbar and inspector controls could be part of it)
